### ISSUE TITLE
ESP32 C3 or S3 Support?

### ISSUE THREAD

**Issue by JayShoe** [pschatzmann/ESP32-A2DP#99]
I'm interested in gaining access to Bluetooth 5 with PHY coded long-range support. What's the likelihood that this library would work with the [C3 ](https://www.espressif.com/en/products/socs/esp32-c3)or [S3](https://www.espressif.com/en/products/socs/esp32-s3)? I have already purchased an [ESP32-c3-devkitc-02](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/hw-reference/esp32c3/user-guide-devkitc-02.html) so I can give it a go. From what I'm reading, I don't think I'll be lucky enough to get "drop in replacement" because the library is new/different. Any thoughts?

**Comment by pschatzmann** [comment]
I just had a quick look at the documentation: We need I2S and Bluetooth Classic A2DP.
The documentation for the C3 seems to be available and identical to the one for the regular ESP32, so there is the chance that this might work on a C3.

Anyhow I would be interested to hear back from you, after you tried it out...

**Comment by pschatzmann** [comment]
Bad news: ESP32-C3 supports Low Energy Bluetooth and Wi-Fi, doesn't support Classic BT.
https://github.com/espressif/esp-idf/issues/6794

**Comment by pschatzmann** [comment]
My mistake the issue relates to the C2 and not the C3.
I filed a new issue https://github.com/espressif/arduino-esp32/issues/5771

**Comment by sathishrajembedded** [comment]
SO, what you are saying??...its shocking!!!!!

ESP32-S3 will not play song via A2DP SINK ??? 

I want to use ESP32-S3 series to stream audio from android bluetooth...Is it possible or not...Please tell me.

Thank you

**Comment by pschatzmann** [comment]
Officially not supported - so it is not possible

**Comment by sathishrajembedded** [comment]
Thank you very much...I planned buy and include in my project. You saved money and time.

Could you tell me the best and latest chip number in ESP32 that supports, audio stream from mobile which Supports APTX, APTXHD .

**Comment by pschatzmann** [comment]
To my knowledge the ESP32 A2DP only supports SBC and I am not aware of any other microcontroller that would support A2DP.

If you need APTX, I suggest that you look into some dedicated A2DP modules or select a more powerfull microcomputer (e.g. Rasperry Pi Zero)

**Comment by sathishrajembedded** [comment]
But, this link
https://github.com/cfint/esp32-a2dp-sink

says that with ESP32 -->Supports aptX, aptX-HD, aptX-LL, LDAC codecs.

based on this , i assumed that with Latest ESP32 with dual core, I thought it is possible

**Comment by pschatzmann** [comment]
Yes, but this is not part of the standard solution provided by Espressif and definitly not supported by this project.

**Comment by sathishrajembedded** [comment]
Okay..... I already using QCC3031-I2S in my product. It is costlier solution in the cost competitive audio market.

So, i searching for low cost solution, that able to receive bluetooth audio from android and give the audio data in I2S 24bit/16bit.

So, any ESP32 or AiThinker module at low cost will do stream A2DP sink with I2S output @16bit ???

**Comment by pschatzmann** [comment]
I suggest to open a separate discussion.  After all this is a closed issue!

**Comment by sathishrajembedded** [comment]
Okay thank you....
